generate_version_translation_unit.sh

```diff
@@ -13,9 +13,17 @@ char const Version[] =
 }  // namespace base
 }  // namespace principia"
 
-echo -e "$VERSION_TEMPLATE" | 
-sed "s/%%DATE%%/`date -d $(git log -1 --format=%cd --date=iso-strict) -u +%Y-%m-%dT%H:%M:%SZ`/" |
-sed "s/%%VERSION%%/`git describe --tags --always --dirty --abbrev=40 --long`/" > $TEMPORARY_FILE
+PLATFORM=$(uname -s)
+if [ "$PLATFORM" == "Darwin" ]; then
+  echo -e "$VERSION_TEMPLATE" |
+  sed "s/%%DATE%%/`date -j -u -f \"%Y-%m-%d %H:%M:%S %z\" \"$(git log -1 --format=%cd --date=iso)\"`/" |
```

Comment:
The format seems inconsistent with the the one we are using, which is really ISO 8601 extended format, that is `%Y-%m-%dT%H:%M:%SZ`; note the T, and note the timezone written simply as Z (with no space) since we give UTC. 
Example in our Catalan build:
````cpp
char const BuildDate[] = "2017-05-26T12:44:02Z";